storage/innobase/lock/lock0lock.cc

```diff
@@ -4466,6 +4532,20 @@ void lock_release_on_prepare(trx_t *trx)
     if (lock_release_on_prepare_try(trx))
       return;
 
+  mtr_t mtr;
+  rec_offs offsets_[REC_OFFS_NORMAL_SIZE];
+  rec_offs *offsets= offsets_;
+  mem_heap_t *heap= nullptr;
+  if (false)
```

Comment:
Why do we need this? func_exit label is used only at the end of the function. Why just don't move the code to the end of the function?

Reply:
I did not want to change the indentation of the rest of the function to ensure that `LockMutexGuard` is released before the conditional call to `mem_heap_free(heap)`. At the end when we do `goto func_exit` to this `if (false)` block, `LockMutexGuard g` will have been destroyed and the latch released.

Reply:
Maybe we could add comment about LockMutexGuard destructor call before "if (false)" condition. Otherwise it looks confusing.

Reply:
I could also just add those `{` and `}` and reindent the code block.